code/go/0chain.net/core/util/mpt_pnodedb_test.go

```diff
@@ -0,0 +1,339 @@
+package util
+
+import (
+	"context"
+	"os"
+	"reflect"
+	"strconv"
+	"testing"
+
+	"github.com/0chain/gorocksdb"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/rand"
+)
+
+const dataDir = "tmp"
+
+func cleanUp() error {
+	if err := os.RemoveAll(dataDir); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func TestNewPNodeDB(t *testing.T) {
+	sstType = SSTTypePlainTable
+
+	type args struct {
+		dataDir string
+		logDir  string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name:    "Test_NewPNodeDB_OK",
+			args:    args{dataDir: dataDir},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := NewPNodeDB(tt.args.dataDir, tt.args.logDir)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("NewPNodeDB() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			if err := cleanUp(); err != nil {
+				t.Error(err)
+			}
+		})
+	}
+}
+
+func TestPNodeDB_Iterate(t *testing.T) {
+	db, cleanUp := newPNodeDB(t)
+	defer cleanUp()
+
+	wo := gorocksdb.NewDefaultWriteOptions()
+	err := db.db.Put(wo, []byte("key"), make([]byte, 0))
+	require.NoError(t, err)
+
+	type fields struct {
+		dataDir  string
+		db       *gorocksdb.DB
+		ro       *gorocksdb.ReadOptions
+		wo       *gorocksdb.WriteOptions
+		to       *gorocksdb.TransactionOptions
+		fo       *gorocksdb.FlushOptions
+		version  int64
+		versions []int64
+	}
+	type args struct {
+		ctx     context.Context
+		handler NodeDBIteratorHandler
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name:    "Test_PNodeDB_Iterate_Err_Reading_OK",
+			fields:  fields{db: db.db},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			pndb := &PNodeDB{
```

Comment:
Move the fields defention inside the t.Run goroutine. The database needs to be setup inside the goroutine to avoid errors with `t.Parallel. See  `Test_PNodeDB_PruneBelowVersion_OK` for example.

Reply:
Done.